File: python/money.py
class Money:
    def __init__(self, amount, currency) -> None:
        self.amount = amount
        self.currency = currency

    def __eq__(self, __o: object) -> bool:
        return self.amount == __o.amount and self.currency == __o.currency

    def times(self, multiplier):
        return Money(self.amount * multiplier, self.currency)

    def divide(self, divisor):
        return Money(self.amount / divisor, self.currency)

File: dart/lib/money.dart
class Money {
  double amount;
  String currency;

  Money(this.amount, this.currency);

  Money times(double multiplier) {
    return Money(multiplier * amount, currency);
  }

  Money divide(double divisor) {
    return Money(amount / divisor, currency);
  }

  @override
  bool operator ==(other) {
    return (other is Money) &&
        other.amount == amount &&
        other.currency == currency;
  }

  @override
  int get hashCode => amount.hashCode ^ currency.hashCode;
}
